src/models/Review.ts

```diff
@@ -0,0 +1,30 @@
+import mongoose, { DateSchemaDefinition, Types } from "mongoose";
+import { User } from "./UserModel";
+import { Post } from "./";
+
+export interface Review extends mongoose.Document {
+    userID: mongoose.Types.ObjectId;
+    postID: mongoose.Types.ObjectId;
+    content: string;
+    timestamp: Date;
+    childrenReview: Types.DocumentArray<Review>;
+    parentReview: {
+        parentReviewID: mongoose.Types.ObjectId | null;
+        childrenReviews: Types.DocumentArray<Review>;
+    };
+}
+
+type ReviewModel = mongoose.Model<Review>;
+const ReviewSchema = new mongoose.Schema<Review, ReviewModel>({
+    userID: { type: mongoose.Types.ObjectId, ref: "User", required: true },
+    postID: { type: mongoose.Types.ObjectId, ref: "Post", required: true },
+    content: String,
+    timestamp: { type: Date, default: Date.now },
+    childrenReview: [{ type: mongoose.Schema.Types.ObjectId, ref: "Review" }],
+    parentReview: {
+        parentReviewID: { type: mongoose.Schema.Types.ObjectId, ref: "Review" },
+        childrenReviews: [{ type: mongoose.Schema.Types.ObjectId, ref: "Review" }],
```

Comment:
You don't need this line. The fact that you are already have the parent Review point to a Review model. It is already recursive.